src/journal-remote/journal-remote-main.c

```diff
@@ -175,7 +180,25 @@ static int request_meta(void **connection_cls, int fd, char *hostname, Compressi
 
         log_debug("Added RemoteSource as connection metadata %p", source);
 
-        source->compression = compression;
+        for (int i = 0; arg_compression != NULL && arg_compression[i] != NULL; i++)
```

Comment:
we typically rely on C's downgrade-to-bool feature when doing pointer null/non-null checks. hence drop the `!= NULL` here and everywhere.

if you iterate through memory objects, please use size_t as counter type